File: platform-demos/C/samples/paned.js
#!/usr/bin/gjs

const Gtk = imports.gi.Gtk;
const Lang = imports.lang;

const PanedExample = new Lang.Class ({
    Name: 'Paned Example',

    // Create the application itself
    _init: function () {
        this.application = new Gtk.Application({ application_id: 'org.example.panedexample' });

       // Connect 'activate' and 'startup' signals to the callback functions
        this.application.connect('activate', Lang.bind(this, this._onActivate));
        this.application.connect('startup', Lang.bind(this, this._onStartup));
    },

    // Callback function for 'activate' signal presents windows when active
    _onActivate: function() {
        this.window.present();
    },

    // Callback function for 'startup' signal builds the UI
    _onStartup: function () {
        this._buildUI ();
    },

    // Build the application's UI
    _buildUI: function() {
        // Create the application window
        this.window = new Gtk.ApplicationWindow  ({ application: this.application,
                                                    window_position: Gtk.WindowPosition.CENTER,
                                                    title: "Paned Window Example",
                                                    default_width: 450,
                                                    default_height: 350,
                                                    border_width: 10 });

        // a new widget with two adjustable panes,
        // one on the left and one on the right
        this.paned = Gtk.Paned.new(Gtk.Orientation.HORIZONTAL);

        // two images
        this.image1 = new Gtk.Image();
        this.image1.set_from_file("gnome-image.png");
        this.image2 = new Gtk.Image();
        this.image2.set_from_file("tux.png");

        // add the first image to the left pane
        this.paned.add1(this.image1);
        // add the second image to the right pane
        this.paned.add2(this.image2)

        // add the panes to the window
        this.window.add(this.paned)
        this.window.show_all();
    }
});

// Run the application
let app = new PanedExample();
app.application.run (ARGV);

File: platform-demos/C/samples/paned.py
from gi.repository import Gtk
import sys


class MyWindow(Gtk.ApplicationWindow):

    def __init__(self, app):
        Gtk.Window.__init__(self, title="Paned Example", application=app)
        self.set_default_size(450, 350)

        # a new widget with two adjustable panes,
        # one on the left and one on the right
        paned = Gtk.Paned.new(Gtk.Orientation.HORIZONTAL)

        # two images
        image1 = Gtk.Image()
        image1.set_from_file("gnome-image.png")
        image2 = Gtk.Image()
        image2.set_from_file("tux.png")

        # add the first image to the left pane
        paned.add1(image1)
        # add the second image to the right pane
        paned.add2(image2)

        # add the panes to the window
        self.add(paned)


class MyApplication(Gtk.Application):

    def __init__(self):
        Gtk.Application.__init__(self)

    def do_activate(self):
        win = MyWindow(self)
        win.show_all()

    def do_startup(self):
        Gtk.Application.do_startup(self)

app = MyApplication()
exit_status = app.run(sys.argv)
sys.exit(exit_status)
